pkg/skaffold/cluster/minikube.go

```diff
@@ -162,21 +130,28 @@ func matchServerURLFor(kubeContext string, serverURL *url.URL) (bool, error) {
 		return false, fmt.Errorf("failed to unmarshal data: %w", err)
 	}
 
+	vmTypeDrivers := map[string]bool{
+		"virtualbox": true,
+		"hyperkit":   true,
+	}
+
+	serverURL, err := url.Parse(server)
+	if err != nil {
+		logrus.Tracef("invalid server url: %v", err)
+	}
+
 	for _, v := range data.Valid {
-		if v.Config.Name != kubeContext {
-			continue
+		for _, n := range v.Config.Nodes {
+			if err == nil && serverURL.Host == fmt.Sprintf("%s:%d", n.IP, n.Port) {
+				return true, nil
+			}
 		}
 
-		if v.Config.Driver != HyperKit && v.Config.Driver != VirtualBox {
+		if !vmTypeDrivers[v.Config.Driver] && v.Config.Name == kubeContext {
```

Comment:
I have to say that I'm not a fan of embedding rather esoteric knowledge of minikube VM drivers here.  On my personal machine, for example, I use the VMware driver and I *think* this case applies there too.

Reply:
I've removed all this now